templates/snap-details.html

```diff
@@ -133,7 +133,15 @@ <h4>Where people are using {{ snap_title }}</h4>
     }).install();
     Raven.context(function () {
       snapcraft.public.screenshots('#js-snap-screenshots');
-      snapcraft.public.map('#js-snap-map', {{ countries|tojson }});
+      try {
+        var mapHolder = document.getElementById('js-snap-map');
+        if (mapHolder) {
+          snapcraft.public.map(mapHolder, {{ countries | tojson }});
+        }
+      } catch(e) {
+        Raven.captureException(e);
+        document.querySelector('.js-snap-map-holder').style.display = 'none';
```

Comment:
I think it may be possible for this line to cause error when there is no`js-snap-map-holder`.

If `countries` is empty, map container would not be rendered in HTML.
So rendering map will fail, but also there would be nothing to set display none on.

Ideally we could possibly not render the piece of JS that tries to render the map if we know that it's disabled.

Reply:
Fixed! Thanks @bartaz